code/modules/surgery/organs/organ_external.dm

```diff
@@ -108,28 +111,33 @@
 				return
 	..()
 
+
 /obj/item/organ/external/update_health()
 	damage = min(max_damage, (brute_dam + burn_dam))
 	return
 
 
-/obj/item/organ/external/New(var/mob/living/carbon/holder, var/internal)
+/obj/item/organ/external/New(var/mob/living/carbon/holder)
 	..()
-	if(owner)
-		replaced(owner)
-		sync_colour_to_human(owner)
+	if(istype(holder, /mob/living/carbon/human))
+		replaced(holder)
+		sync_colour_to_human(holder)
 	spawn(1)
 		get_icon()
 
 /obj/item/organ/external/replaced(var/mob/living/carbon/human/target)
 	owner = target
 	status = status & ~ORGAN_DESTROYED
+	forceMove(owner)
 	if(istype(owner))
 		owner.organs_by_name[limb_name] = src
 		owner.organs |= src
 		for(var/obj/item/organ/organ in src)
-			organ.loc = owner
-			organ.replaced(owner,src)
+			if(istype(src, /obj/item/organ/internal))
+				var/obj/item/organ/internal/I = organ
+				if(target.get_organ_slot(I.slot))
+					continue // Just leave it inside its limb, so brains with brainmobs in them don't get voided.
+				organ.replaced(owner,src)
```

Comment:
This replaced should be outside of the if statement, otherwise hands will remain inside arms when you stick them back on.